### DISCUSSION TITLE
Bindings when we have a View State

### DISCUSSION THREAD

**Discussion by hadiidbouk** [pointfreeco/swift-composable-architecture#820] [Ideas]
Following the conversation on Twitter [here](https://twitter.com/DboukHadi/status/1438271065139122181?s=20), I found that we will face some complexity and more code to write when we use View states alongside the new binding way.

I am asking if it will be possible to make the `@BindableState` properties inside a protocol, and this protocol will be used in the State and the View State?

**Comment by tgrapperon** [comment]
Without contributing to this discussion yet, I'm just linking the other discussion #769 related to this topic for the sake of completeness.

**Comment by iampatbrown** [comment]
@hadiidbouk Do you have the before `@BindableState` version of that example as well? I'd like to explore this a little further and test out a few of @tgrapperon's ideas.
